### PR TITLE
Create buttons for Seen and Crafting

### PR DESCRIPTION
**Pull request by elShailee**

### CODE REVIEW

File: src/Components/AllTheFloatingWindows/AllTheFloatingWindows.js [hunk 1]
```diff
@@ -2,13 +2,17 @@ import React from 'react';
...
 export default function AllTheFloatingWindows() {
 	const dispatch = useDispatch();
 
 	const isCoinsPurchaseWindowMounted = useSelector(state => state.MountedComponents.CoinsPurchaseWindow);
 	const isEnergyPurchaseWindowMounted = useSelector(state => state.MountedComponents.EnergyPurchaseWindow);
+	const isCraftingWindowMounted = useSelector(state => state.MountedComponents.CraftingWindow);
+	const isSeenWindowMounted = useSelector(state => state.MountedComponents.SeenWindow);
 	const isActionsWindowMounted = useSelector(state => state.MountedComponents.ActionsWindow);
 
 	const CoinsPurchaseWindowAsVar = (
```
Comment (elShailee) [inline src/Components/AllTheFloatingWindows/AllTheFloatingWindows.js:18]: please maintain a consistent order in the components, to maintain readability

File: src/Components/Crafting/CraftingButton.js [hunk 2]
```diff
@@ -1,7 +1,29 @@
...
+			dispatch(mountComponents('CraftingWindow'));
+		}
+	};
+
+	return (
+		<div id='craftingButtonContainer'>
+			<GeneralButton id='craftingButton' uppercased outlined onButtonClick={onCraftingButtonClick}>
+				{craftingButtonTexts.title}
+			</GeneralButton>
+			{isCraftingWindowOpen && <CraftingWindow />}
```
Comment (elShailee) [inline src/Components/Crafting/CraftingButton.js:26]: please remove, not needed

File: src/Components/Crafting/CraftingButton.js [hunk 3]
```diff
@@ -1,7 +1,29 @@
...
 import './CraftingButton.css';
+import GeneralButton from 'Components/GeneralComponents/GeneralButton';
 import { craftingButtonTexts } from 'Texts/gameplayTexts';
+import { mountComponents, unmountComponents } from 'Redux/Slices/MountedComponents'; //???
+import { useDispatch, useSelector } from 'react-redux';
 
 export default function CraftingButton() {
-  return <div id="craftingButton">{craftingButtonTexts.title}</div>;
+	const dispatch = useDispatch();
+	const isCraftingWindowOpen = useSelector(state => state.MountedComponents.CraftingWindow);
```
Comment (elShailee) [inline src/Components/Crafting/CraftingButton.js:11]: please change to `isCraftingWindowMounted`

File: src/Components/Seen/SeenButton.js [hunk 4]
```diff
@@ -1,7 +1,29 @@
...
 import './SeenButton.css';
+import GeneralButton from 'Components/GeneralComponents/GeneralButton';
 import { seenButtonTexts } from 'Texts/gameplayTexts';
+import { mountComponents, unmountComponents } from 'Redux/Slices/MountedComponents';
+import { useDispatch, useSelector } from 'react-redux';
 
 export default function SeenButton() {
-  return <div id="seenButton">{seenButtonTexts.title}</div>;
+	const dispatch = useDispatch();
+	const isSeenWindowOpen = useSelector(state => state.MountedComponents.SeenWindow);
```
Comment (elShailee) [inline src/Components/Seen/SeenButton.js:11]: please change `open` to `Mounted`

File: src/Components/Seen/SeenButton.js [hunk 5]
```diff
@@ -1,7 +1,29 @@
...
+			dispatch(mountComponents('SeenWindow'));
+		}
+	};
+
+	return (
+		<div id='seenButtonContainer'>
+			<GeneralButton id='seenButton' uppercased outlined onButtonClick={onSeenButtonClick}>
+				{seenButtonTexts.title}
+			</GeneralButton>
+			{isSeenWindowOpen && <SeenWindow />}
```
Comment (elShailee) [inline src/Components/Seen/SeenButton.js:26]: please remove

File: src/Components/Crafting/CraftingWindow.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import GeneralWindow from 'Components/GeneralComponents/GeneralWindow';
+
+export default function CraftingWindow({ unmountCraftingWindow }) {
+	return (
+		<GeneralWindow unmountGeneralWindow={unmountCraftingWindow} rightSided={true}>
```
Comment (elShailee) [inline src/Components/Crafting/CraftingWindow.js:6]: please change `rightSided={true}` to `rightSided`

File: src/Components/Seen/SeenWindow.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import GeneralWindow from 'Components/GeneralComponents/GeneralWindow';
+
+export default function SeenWindow({ unmountSeenWindow }) {
+	return (
+		<GeneralWindow unmountGeneralWindow={unmountSeenWindow} rightSided={true}>
```
Comment (elShailee) [inline src/Components/Seen/SeenWindow.js:6]: please change rightSided={true} to rightSided

File: src/Redux/Slices/MountedComponents.js [hunk 8]
```diff
@@ -9,6 +9,8 @@ const initialState = {
 	CoinsPurchaseWindow: false,
 	CoinsTab: false,
 	EnergyTab: false,
+	Crafting: false,
+	Seen: false,
```
Comment (elShailee) [inline src/Redux/Slices/MountedComponents.js:13]: Please change `Cafting` and `Seen` keys to `CraftingWindow` and `SeenWindow`.

File: src/Components/AllTheFloatingWindows/AllTheFloatingWindows.js (hunk 1 above)
Comment (BorodulinRoman) [reply src/Components/AllTheFloatingWindows/AllTheFloatingWindows.js:18]: Done

File: src/Components/Crafting/CraftingButton.js (hunk 2 above)
Comment (BorodulinRoman) [reply src/Components/Crafting/CraftingButton.js:26]: Done.

File: src/Components/Crafting/CraftingButton.js (hunk 3 above)
Comment (BorodulinRoman) [reply src/Components/Crafting/CraftingButton.js:11]: Done.

File: src/Components/Seen/SeenButton.js (hunk 4 above)
Comment (BorodulinRoman) [reply src/Components/Seen/SeenButton.js:11]: Done.

File: src/Components/Seen/SeenButton.js (hunk 5 above)
Comment (BorodulinRoman) [reply src/Components/Seen/SeenButton.js:26]: Done.

File: src/Components/Crafting/CraftingWindow.js (hunk 6 above)
Comment (BorodulinRoman) [reply src/Components/Crafting/CraftingWindow.js:6]: Done.

File: src/Components/Seen/SeenWindow.js (hunk 7 above)
Comment (BorodulinRoman) [reply src/Components/Seen/SeenWindow.js:6]: Done.

File: src/Redux/Slices/MountedComponents.js (hunk 8 above)
Comment (BorodulinRoman) [reply src/Redux/Slices/MountedComponents.js:13]: Done.

File: src/Screens/Gameplay/Components/Crafting/CraftingButton.jsx [hunk 9]
```diff
@@ -1,7 +1,27 @@
...
+			dispatch(mountComponents('CraftingWindow'));
+		}
+	};
+
+	return (
+		<div id='craftingButtonContainer'>
+			<GeneralButton id='craftingButton' uppercased outlined onButtonClick={onCraftingButtonClick}>
+				{texts.gameplay.crafting.buttonTitle}
+			</GeneralButton>
+		</div>
```
Comment (elShailee) [inline src/Screens/Gameplay/Components/Crafting/CraftingButton.jsx:25]: I believe there's no need for the container div.

File: src/Screens/Gameplay/Components/Seen/SeenButton.jsx [hunk 10]
```diff
@@ -1,7 +1,27 @@
...
+			dispatch(mountComponents('SeenWindow'));
+		}
+	};
+
+	return (
+		<div id='seenButtonContainer'>
+			<GeneralButton id='seenButton' uppercased outlined onButtonClick={onSeenButtonClick}>
+				{texts.gameplay.seen.buttonTitle}
+			</GeneralButton>
+		</div>
```
Comment (elShailee) [inline src/Screens/Gameplay/Components/Seen/SeenButton.jsx:25]: same here - I believe there's no need for the container div.

**Comment by elShailee** [comment]
also - there's a `text-align: center;` property at both the button's CSS files that is irrelevant because of the styling of `GeneralButton`.

overall - looks great!
send me a WhatsApp message when you're done (and of course - If you have any questions or disagreements about these notes).

have you found yourself another task?

**Review by odeyah — APPROVED** [review]

**Pull request merged**